src/components/password-input/index.md

```diff
@@ -0,0 +1,77 @@
+---
+title: Password input
+description: Help users accessibly enter passwords
+section: Components
+aliases: pass word, pass phrase
+backlogIssueId: 240
+layout: layout-pane.njk
+---
+
+Help users to accessibly create and enter passwords.
+
+## When to use this component
+
+You should use this component whenever you need users to create or enter a password.
+
+Before using this pattern, you should also read the guidance on [asking users for passwords](/patterns/passwords/) and [creating user accounts](/patterns/create-accounts/).
+
+## When not to use this component
+
+Do not use this component if the information being asked for isn't a password or other sensitive security information. Use a normal [text input](/components/text-input/) instead.
+
+## How it works
+
+All password inputs must have labels, and in most cases the label should be visible.
+
+You should align labels above the password input they refer to. They should be short, direct and written in sentence case. Do not use colons at the end of labels.
+
+### Error messages
+
+[TODO]
+
+- Do not indicate whether the username or password was incorrect, avoid brute forcing data
+- Should clear the password input rather than autofilling with the failed value
+- How errors should work if two password fields need to match and don't
+
+### Hide passwords by default
+
+Users might be in a public space when entering or creating a password, so you should hide passwords by default.
+
+When there are two or more password inputs on a page, the labels and 'show' and 'hide' toggles for each password input must be different.
+
+For example, you can label the input "Password" as "show password" and label the second input "Re-enter password" as "show re-entered password".
+
+### Use the autocomplete attribute
+
+Use the `autocomplete` attribute on password inputs to help users complete forms more quickly. This lets you indicate to browsers and password managers what kind of password is needed so they can autofill the information on a user’s behalf.
+
+Set the `autocomplete` attribute to `new-password` if the user is creating a password. Otherwise, use `current-password`.
+
+Providing an `autocomplete` attribute is required to meet [WCAG 2.2](https://www.w3.org/TR/WCAG22/)'s [accessible authentication](https://www.w3.org/TR/WCAG22/#accessible-authentication-minimum) criterion.
+
+Note that even if `autocomplete` is omitted or set to `off`, many browsers will ignore this and continue to autofill password inputs.
+
+### Allow users to paste their password
+
+Do not disable paste on password fields. People may have very good reasons why they want to paste their password, for example if they’re using a password manager.
+
+### Avoid restricting the user's input
+
+[TODO — maybe just keep this on pattern guidance page?]
```

Comment:
Might want to explicitly mention that the field should not hardcode a length limit (truncation), even if there is a backend limit - that instead the user's input should be accepted by the input, and a regular validation error thrown.

Reply:
Context: the guidance is to not limit password lengths, but many legacy systems have them as a requirement. Teams working in such legacy environments may naturally set a max-length on the input thinking they are being helpful, and unaware of the usability issues they are introducing.

Reply:
I've added a paragraph to that effect